internal/rootcoord/root_coord.go

```diff
@@ -2283,17 +2283,22 @@ func (c *Core) Import(ctx context.Context, req *milvuspb.ImportRequest) (*milvus
 
 	// Get collection/partition ID from collection/partition name.
 	var cID int64
-	var ok bool
-	c.MetaTable.ddLock.RLock()
-	defer c.MetaTable.ddLock.RUnlock()
-	if cID, ok = c.MetaTable.collName2ID[req.GetCollectionName()]; !ok {
-		log.Error("failed to find collection ID for collection name",
-			zap.String("collection name", req.GetCollectionName()))
-		return nil, fmt.Errorf("collection ID not found for collection name %s", req.GetCollectionName())
-	}
 	var pID int64
-	var err error
-	if pID, err = c.MetaTable.getPartitionByName(cID, req.GetPartitionName(), 0); err != nil {
+	var ok bool
+	if err := func() error {
+		c.MetaTable.ddLock.RLock()
+		defer c.MetaTable.ddLock.RUnlock()
+		if cID, ok = c.MetaTable.collName2ID[req.GetCollectionName()]; !ok {
+			log.Error("failed to find collection ID for collection name",
+				zap.String("collection name", req.GetCollectionName()))
+			return fmt.Errorf("collection ID not found for collection name %s", req.GetCollectionName())
+		}
+		var err error
+		if pID, err = c.MetaTable.getPartitionByName(cID, req.GetPartitionName(), 0); err != nil {
+			return err
+		}
+		return nil
```

Comment:
Why not extract these code to a validateXX function?

Reply:
I have reorganized the code. PTAL.